plugins/inputs/mesos/mesos_test.go

```diff
@@ -683,12 +683,66 @@ func TestCreateHTTPClientIAM(t *testing.T) {
 	require.NotNil(client)
 	require.NotNil(client.Transport)
 
-	// test that all requests on client send an Authorization header with token
+	// test that all requests on client send an Authorization header with token and default User-Agent header
 	tokenTestServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		auth := r.Header.Get("Authorization")
 		if auth != "token="+testToken {
 			t.Fatalf("Expected request header: `Authorization: token=%s`. Got: %s", testToken, auth)
 		}
+		userAgent := r.Header.Get("User-Agent")
+		if userAgent != "dcos-go" {
+			t.Fatalf("Expected request header: `User-Agent: dcos-go`. Got: %s", userAgent)
+		}
+	}))
+	defer tokenTestServer.Close()
+
+	c := http.Client{
+		Transport: client.Transport,
+	}
+
+	resp, err := c.Get(tokenTestServer.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	resp.Body.Close()
+}
+
+func TestCreateHTTPClientUserAgent(t *testing.T) {
```

Comment:
This new test is almost entirely a duplicate of `TestCreateHTTPClientIAM()` -- the only differences are providing the user-agent to the `Mesos` struct and the user-agent assertion in the server's request handler. This suggests the tests ought to be refactored.